probe/endpoint/ebpf.go

```diff
@@ -287,10 +310,33 @@ func (t *EbpfTracker) isDead() bool {
 
 func (t *EbpfTracker) stop() {
 	t.Lock()
+	defer t.Unlock()
+
 	alreadyDead := t.dead
 	t.dead = true
-	t.Unlock()
 	if !alreadyDead && t.tracer != nil {
 		t.tracer.Stop()
 	}
 }
+
+func (t *EbpfTracker) restart() error {
+	t.Lock()
+	defer t.Unlock()
+
+	t.dead = false
+	t.ready = false
+
+	t.openConnections = map[fourTuple]ebpfConnection{}
+	t.closedDuringInit = map[fourTuple]struct{}{}
+	t.closedConnections = []ebpfConnection{}
+
+	tracer, err := tracer.NewTracer(t)
+	if err != nil {
+		return err
+	}
+
+	t.tracer = tracer
+	tracer.Start()
+
+	return nil
+}
```

Comment:
I'd be wary of holding the lock while calling `tracer.Start()`.

Also, `newEbpfTracker()` should be refactored to call this.